lib/node_modules/@stdlib/array/base/reshape/docs/types/test.ts

```diff
@@ -0,0 +1,97 @@
+/*
+* @license Apache-2.0
+*
+* Copyright (c) 2025 The Stdlib Authors.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+import reshape = require( './index' );
+
+
+// TESTS //
+
+// The function returns an array...
+{
+	const x = [ [ 1, 2 ], [ 3, 4 ] ];
+
+	reshape<number>( x, [ 2, 2 ], [ 4, 1 ], false ); // $ExpectType number[]
```

Comment:
Is it necessary to explicitly use `<number>` here? 

Same thing for `<string>` below. I would think that the compiler is smart enough to infer.